interaction-core/src/test/java/com/temenos/interaction/core/rim/TestHTTPHypermediaRIM.java

```
@@ -243,6 +262,58 @@ public void testDecodePathParametersPercent() throws InteractionException {
         verify(mockCommand).execute((InteractionContext) argThat(new InteractionContextArgumentPathMatcherPercent()));
     }
+    
+    
+    
+    /*
+     * testing method extractDecodedUriSegment 
+     * 
+     * 
+     */
+    @Test
+    public void testExtractDecodedUriSegments() throws InteractionException {
+        ResourceState initialState = new ResourceState("entity", "state", mockActions(), "/test");
+        InteractionCommand mockCommand = mock(InteractionCommand.class);
+        when(mockCommand.execute(any(InteractionContext.class))).thenReturn(Result.FAILURE);
+        HTTPHypermediaRIM rim = new HTTPHypermediaRIM(mockCommandController(mockCommand), new ResourceStateMachine(
+                initialState), createMockMetadata());
+
+        UriInfo uriInfo = mock(UriInfo.class);
+        when(uriInfo.getPath(anyBoolean())).thenReturn("notes/id('id%2FwithSlash')/view");
+        String[] expectedResultArray = new String[]{"notes", "id('id/withSlash')", "view"};
+        
+        String[] resultArray =  rim.extractDecodedUriSegments(uriInfo);
+        assertEquals(3, resultArray.length);
+        assertArrayEquals(expectedResultArray, resultArray);
+    }
+    
+    /*
+     * testing method extractDecodedUriSegment with null and empty input
+     * 
+     * 
+     */
+    @Test
+    public void testExtractDecodedUriSegmentsWithNullAndEmptyInput() throws InteractionException {
+        ResourceState initialState = new ResourceState("entity", "state", mockActions(), "/test");
+        InteractionCommand mockCommand = mock(InteractionCommand.class);
+        when(mockCommand.execute(any(InteractionContext.class))).thenReturn(Result.FAILURE);
+        HTTPHypermediaRIM rim = new HTTPHypermediaRIM(mockCommandController(mockCommand), new ResourceStateMachine(
+                initialState), createMockMetadata());
+        String[] resultArray =  rim.extractDecodedUriSegments(null);
+        if(resultArray!= null) {
+            fail("");
```

Add a failure message instead of empty string